pegasus/src/social_metadata.rb

```diff
@@ -0,0 +1,160 @@
+        image: {path: "/images/cs-is-everything-thumbnail.png", width: 1200, height: 627},
+        video: {youtube_key: "QvyTEx1wyOY", width: 640, height: 360}
+      }
+    },
+    "hourofcode.com" => {
+      "soon-hoc" => {
+        title: I18n.t(:og_title_soon),
+        description: hoc_s(:meta_tag_og_description),
+        description_twitter: hoc_s(:meta_tag_twitter_description),
+        image: {path: "/images/social-media/hourofcode-2015-video-thumbnail.jpg", width: 1440, height: 900},
+        video: {youtube_key: "KsOIlDT145A", width: 640, height: 360}
+      },
+      "week-before-hoc" => {
+        title: I18n.t(:og_title_soon),
+        description: hoc_s(:meta_tag_og_description),
+        description_twitter: hoc_s(:meta_tag_twitter_description),
+        image: {path: "/images/social-media/hourofcode-2015-video-thumbnail.jpg", width: 1440, height: 900},
+        video: {youtube_key: "KsOIlDT145A", width: 640, height: 360}
+      },
+      "actual-hoc" => {
+        title: I18n.t(:og_title_here),
+        description: hoc_s(:meta_tag_og_description),
+        description_twitter: hoc_s(:meta_tag_twitter_description),
+        image: {path: "/images/social-media/hourofcode-2015-video-thumbnail.jpg", width: 1440, height: 900},
+        video: {youtube_key: "KsOIlDT145A", width: 640, height: 360}
+      },
+      "default" => {
+        title: hoc_s(:meta_tag_og_title),
+        description: hoc_s(:meta_tag_og_description),
+        description_twitter: hoc_s(:meta_tag_twitter_description),
+        image: {path: "/images/social-media/hourofcode-2015-video-thumbnail.jpg", width: 1440, height: 900},
+        video: {youtube_key: "KsOIlDT145A", width: 640, height: 360}
+      }
+    },
+    "challenge" => {
+      "soon-hoc" => {
+        title: "Celebrity Challenge",
+        description: "Win a celebrity video chat for your class!",
+        image: {path: "/images/fit-1920/social-media/hoc-student-challenge.jpg", width: 1920, height: 1080}
+      },
+      "week-before-hoc" => {
+        title: "Celebrity Challenge",
+        description: "Win a celebrity video chat for your class!",
+        image: {path: "/images/fit-1920/social-media/hoc-student-challenge.jpg", width: 1920, height: 1080}
+      },
+      "actual-hoc" => {
+        title: "Celebrity Challenge",
+        description: "Win a celebrity video chat for your class!",
+        image: {path: "/images/fit-1220/social-media/celeb-challenge.jpg", width: 1220, height: 640}
+      },
+      "default" => {
+        title: "Celebrity Challenge",
+        description: "Win a celebrity video chat for your class!",
+        image: {path: "/images/fit-1220/social-media/celeb-challenge.jpg", width: 1220, height: 640}
+      }
+    }
+  }
+
+  if request.path == "/challenge" && request.site == "code.org"
+    page = "challenge"
+  elsif request.path == "/" && ["code.org", "csedweek.org", "hourofcode.com"].include?(request.site)
+    page = request.site
```

Comment:
Nit: any reason to use string keys in the `social_tags` hash vs. using `request.site.to_sym` here?

(Same feedback for `hoc_mode` below.)

Reply:
I did try it that way, but figured it functioned more naturally as a string than a symbol.  (Though maybe there's some Ruby rule of thumb I don't know.)

Reply:
Yeah, good point — since the strings contain special characters I don't have a strong preference either way.  I think it's fine as written.